apps/server/src/modules/tool/tool-launch/tool-launch.module.ts

```diff
@@ -18,7 +18,7 @@ import { BasicToolLaunchStrategy, Lti11ToolLaunchStrategy, OAuth2ToolLaunchStrat
 		ContextExternalToolModule,
 		LegacySchoolModule,
 		UserModule,
-		PseudonymModule,
+		forwardRef(() => PseudonymModule), // i do not like this solution, the root problem is on other place but not detectable for me
```

Comment:
I get an error without forwadRef. But will check it again.

Reply:
Error: Nest cannot create the ToolLaunchModule instance.
The module at index [6] of the ToolLaunchModule "imports" array is undefined.

Potential causes:
- A circular dependency between modules. Use forwardRef() to avoid it. Read more: https://docs.nestjs.com/fundamentals/circular-dependency
- The module at index [6] is of type "undefined". Check your import statements and the type of the module.

Scope [ServerModule -> CollaborativeStorageModule -> CollaborativeStorageAdapterModule -> ToolModule]


If you help me to find the root i am happy. But i has no time to get deep into debugging this issue. The error message bring me not to the point that i can fix this correctly.

Reply:
Any idea? I am happy if i can avoid this.

Reply:
Didnt had any time to check this. forwardRef should be the last option.. if it is then fine for me

Reply:
Yes i think in any case we have is forwardRef only a dependency is wrong issue. But is not easy to solve. I get many many dependency issues in the PR and by fixing it the changed file size of the PR is increase by +200%. If i have an idea how i get fix this i can do it also. But really no idea.

Reply:
Is the forwardRef here still needed after moving the rules?

Reply:
Good point i try to remove it again.

Reply:
No sorry, error throw again. The root evil for this must be on other place.